packages/firestore/src/remote/persistent_stream.ts

```diff
@@ -495,6 +492,44 @@ export abstract class PersistentStream<
     // without a backoff accidentally, we set the stream to error in all cases.
     return this.close(PersistentStreamState.Error, error);
   }
+
+  /** Updates this.state and increments our sequence number. */
+  private updateState(newState: PersistentStreamState): void {
+    log.debug(
+      LOG_TAG,
+      `state changed from ${PersistentStreamState[this.state]} to ${
+        PersistentStreamState[newState]
+      }`
+    );
+    this.state = newState;
+    this.stateSequenceNumber++;
+  }
+
+  /**
+   * Returns a "dispatcher" function that dispatches operations onto the
+   * AsyncQueue but only runs them if startSequenceNumber remains unchanged.
+   * This allows us to turn auth / stream callbacks into no-ops if the stream
+   * is closed / re-opened, etc.
+   */
+  private stateGuardedDispatcher(): (fn: () => Promise<void>) => void {
```

Comment:
I think this would be cleaner if it:
- was called something along the lines of `dispatchIfStateUnchanged`
- you passed in the state sequence number

Reply:
`closeGuardedDispatcher` still seems a little off to me. The name is very tied to its implementation, but at first glance it doesn't seem to convey what it is used for (it might also just be the choice of words).

I don't have a good suggestion for you, but even something like `getCloseGuardedDispatcher()` might clear up some of my confusion.

Reply:
getCloseGuardedDispatcher() SGTM